src/main/java/com/nha/abdm/wrapper/hip/hrp/GatewayCallbackController.java

```diff
@@ -183,6 +193,37 @@ public ResponseEntity<GatewayCallbackResponse> onAddCareContext(
     return new ResponseEntity<>(HttpStatus.ACCEPTED);
   }
 
+  @PostMapping({"/v0.5/consents/hip/notify"})
+  public ResponseEntity<GatewayCallbackResponse> hipNotify(
+      @RequestBody HIPNotifyRequest hipNotifyRequest) {
+    if (hipNotifyRequest != null
+        && hipNotifyRequest.getHIPNotification() != null
+        && hipNotifyRequest.getHIPNotification().getConsentDetail() != null
+        && hipNotifyRequest.getHIPNotification().getConsentDetail().getPatient() != null) {
+      patientService.addConsentArtefacts(hipNotifyRequest);
+      HIPOnNotifyRequest hipOnNotifyRequest =
+          HIPOnNotifyRequest.builder()
+              .requestId(hipNotifyRequest.getRequestId())
+              .timestamp(Utils.getCurrentTimeStamp())
+              .acknowledgement(
+                  Acknowledgement.builder()
+                      .consentId(hipNotifyRequest.getHIPNotification().getConsentId())
+                      .status(HttpStatus.OK.name())
+                      .build())
+              .build();
+      consentService.hipOnNotify(hipOnNotifyRequest);
```

Comment:
Shouldn't we first respond back to the client `Gateway` and then initiate the `onNotify` call?

Reply:
I am not sure if the order matters. We can do both in parallel or in any order. The response back is not an new http connection but a response to the running http call. Any business logic I think should be part of this call.

Reply:
This means that the Gateway would get a `on-notify` request before even the `hip` completing the previous `notify` call. This would lead to some confusion at the `Gateway` side.

Reply:
As discussed, I will try to analyse a bit more on this and will respond back. In the meanwhile, approving this to unblock further changes.